### PR TITLE
[rfw] Increase tolerance for material widget tests

### PR DESCRIPTION
**Pull request by victoreronmosele**
This PR introduces a tolerance for rfw material widget tests to fix the golden test error in https://github.com/flutter/flutter/issues/151611.

Part of a fix for https://github.com/flutter/flutter/issues/151659.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style], or this PR is [exempt from CHANGELOG changes].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/blob/master/docs/contributing/Chat.md
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#version
[following repository CHANGELOG style]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog-style
[exempt from CHANGELOG changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog
[test-exempt]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#tests

### CODE REVIEW

File: packages/rfw/test/material_widgets_test.dart [hunk 1]
```diff
@@ -23,6 +25,9 @@ void main() {
   }
 
   testWidgets('Material widgets', (WidgetTester tester) async {
+    setUpTolerantComparator(
+        testPath: 'test/material_widget_test.dart', precisionTolerance: 0.01);
```
Comment (Hixie) [inline packages/rfw/test/material_widgets_test.dart:29]: how much is 0.01?

**Comment by Hixie** [comment]
Do we understand what kinds of changes are causing failures here? Is it anti-aliasing? Blurs? Something else?

File: packages/rfw/test/material_widgets_test.dart (hunk 1 above)
Comment (victoreronmosele) [reply packages/rfw/test/material_widgets_test.dart:29]: `0.01` is `4800px` out of the `480000px` in the `material_test.scaffold.png` file. 

The actual diff from the failing test is `4px` so a more precise value for the tolerance is `0.00000833333`.

**Comment by victoreronmosele** [comment]
> Do we understand what kinds of changes are causing failures here? Is it anti-aliasing? Blurs? Something else?

I can't tell which one it is, but I generated the test failure artifacts in a GitHub Action workflow run [here](https://github.com/victoreronmosele/rfw_golden_tests_workflow/actions/runs/10228220835/artifacts/1772200192). 

---

Also, I pushed an update to the PR to set up the tolerant comparator for all the tests in the file since the failure was occurring in multiple tests.

**Comment by Hixie** [comment]
I see nothing different in those diffs at all. :-/

File: packages/rfw/test/material_widgets_test.dart (hunk 1 above)
Comment (Hixie) [reply packages/rfw/test/material_widgets_test.dart:29]: Meaning there are four pixels that have a non-identical match to the control?

Comment (Hixie) [reply packages/rfw/test/material_widgets_test.dart:29]: I worry that allowing four pixels to be just flat out wrong means that if we start incorrectly rendering a dot somewhere, we won't notice it. "Tolerance" makes it sound like we'd accept it if the four pixels happened to be slightly off in color (e.g. if 0xFFFEFEFE became 0xFFAFAFA or something), not that we'd accept wildly wrong renderings (e.g. 0xFF00FF00 becoming 0xFFFF0000).

Comment (victoreronmosele) [reply packages/rfw/test/material_widgets_test.dart:29]: True. What would be a better fix for the issue? Updating the control images? 

Also, I implemented the tolerance based on this comment in the original issue: https://github.com/flutter/flutter/issues/151611#issuecomment-2223874000.

**Comment by victoreronmosele** [comment]
Cropping and zooming in on the images show some changes in the diffs.

See for `material_test.overflow_bar_resembles_button_bar_masterImage.png`:

| Isolated Diff | Masked Diff |
|---------------|-------------|
| Isolated Diff | Masked Diff |
| **Master Image** | **Test Image** |
| Master Image | Test Image |

Original `material_test.overflow_bar_resembles_button_bar_masterImage.png`:

material_test overflow_bar_resembles_button_bar_masterImage

File: packages/rfw/test/material_widgets_test.dart (hunk 1 above)
Comment (Hixie) [reply packages/rfw/test/material_widgets_test.dart:29]: Updating the control images is a short-term fix, it doesn't really address the underlying issue (which is that skia and impeller don't guarantee pixel-identical rendering across versions for things like antialiasing and blurs, but the test is assuming they do).

I don't have a good solution. Setting the tolerance to something like 0.00002 (so if I understood correctly, allowing about 10 pixels to drift from the control) might be ok I guess? In the absence of any better ideas...

Comment (victoreronmosele) [reply packages/rfw/test/material_widgets_test.dart:29]: Yes, 10px away from the control. 

I'll update the tolerance.

Comment (victoreronmosele) [reply packages/rfw/test/material_widgets_test.dart]: Updated now in 0fd9943.

**Comment by stuartmorgan** [comment]
The referenced issue is closed; is this obsolete, or is changing the allowed diff still something we want to do?

**Comment by victoreronmosele** [comment]
@stuartmorgan, the relevant issue is https://github.com/flutter/flutter/issues/151659. It is still open. 

https://github.com/flutter/flutter/issues/151659 says to fix and enable the rfw tests. 

This PR is doing the fix. 

If this gets merged, another PR should be raised to enable the actual tests in the [flutter/tests](https://github.com/flutter/tests) repo.

**Comment by stuartmorgan** [comment]
From triage: @Hixie, what is the next step here from a reviewer standpoint?

**Review by Hixie — APPROVED** [review]
Sorry, this fell of my radar.

I guess LGTM, but I'm a bit skeptical of this approach. I would prefer to find ways to make Flutter not change rendering across versions (e.g. the way we do for shadows, where in golden tests the shadows become big black rectangles).

**Review by stuartmorgan — APPROVED** [review]

**Pull request merged**